exercises/src/test/scala/state/StateSpec.scala

```diff
@@ -0,0 +1,200 @@
+import org.specs2.mutable.Specification
+import org.specs2.ScalaCheck
+
+import fpinscala.state._
+
+class StateSpec extends Specification with ScalaCheck {
+ 
+ //========================================================================================
+ //#6.1
+  "nonNegativeInt" should {
+    "return a bigger than or equal to zero Int Output along with a new State" in {
+      prop { 
+        x: Int =>
+          val rng1 = new RNG.Simple(x)
+          val (i, rng2) = RNG.nonNegativeInt(rng1) 
+          (i >= 0) mustEqual true
+          rng2 mustNotEqual rng1
+           }
+    }
+  }
+
+ //========================================================================================
+ //#6.2
+  "double" should {
+    "return a bigger than or equal to zero and less than one double Output along with a new State" in {
+      prop { 
+        y: Long =>
+          val rng1 = new RNG.Simple(y)
+          val (d, rng2) = RNG.double(rng1) 
+          d must beBetween(0.0, 1.0).excludingEnd
+          rng1 mustNotEqual rng2
+           }
+    }
+  }
+
+ //========================================================================================
+ //#6.3
+  "intDouble" should {
+    "return a pair of Int and Double Output along with a new State" in {
+      prop { 
+        y: Long =>
+          val rng1 = new RNG.Simple(y)
+          val ((i,d), rng2) = RNG.intDouble(rng1) 
+          d must beBetween(0.0, 1.0).excludingEnd
```

Comment:
Since you're testing this many times (in multiple tests) you could extract it into a method of the `StateSpec` class, like `def isDoubleBetween0And1Ex(d: Double) = d must beBetween(0.0, 1.0).excludingEnd` and then use that everywhere.